### DISCUSSION TITLE
Can't get Intellisense to work

### DISCUSSION THREAD

**Discussion by wintertims** [mark-wiemer/ahkpp#421] [Q&A] [answered]
I recently added this extension to VSCode, and the syntax highlighting looks great. However, I can't seem to the function hover Intellisense to work (for it to show the comment above the function and the function's parameters). The same goes for using ctrl-click on a symbol to go to its definition.

Is there any extra configuration I need to add beyond installing the extension?

**Comment by monkeylander3000** [comment, marked as answer]
Certainly! It’s frustrating when IntelliSense doesn’t work as expected. Let’s troubleshoot this together. Here are some steps you can take to ensure that IntelliSense behaves correctly in Visual Studio Code (VS Code):

Enable Hover:
Open your VS Code settings (you can do this by pressing Ctrl + , or navigating to File > Preferences > Settings).
Search for editor.hover.enabled in the search bar.
Make sure the value is set to true. If not, update it to enable hover functionality:
JSON

{
    "editor.hover.enabled": true
}
AI-generated code. Review and use carefully. [More info on FAQ](https://www.bing.com/new#faq).
[Alternatively, you can use the UI settings by searching for “hover enabled” in the settings and ensuring it’s enabled](https://stackoverflow.com/questions/57770473/how-to-enable-hover-in-vs-code)[1](https://stackoverflow.com/questions/57770473/how-to-enable-hover-in-vs-code).
Check Language Extensions:
IntelliSense is enabled by default for languages like JavaScript and TypeScript.
If you’re working with a different language, make sure you have the relevant language extension installed. You can search for extensions in the VS Code marketplace.
Sometimes, extensions might get disabled accidentally. To fix this:
Go to the Extensions view (you can do this by pressing Ctrl + Shift + X).
Type @builtin in the search bar.
[Scroll to your language extension (e.g., TypeScript and JavaScript) in the Features list and ensure it’s enabled](https://stackoverflow.com/questions/57770473/how-to-enable-hover-in-vs-code)[2](https://www.thewindowsclub.com/intellisense-not-working-in-vs-code).
Restart VS Code:
Sometimes a simple restart can resolve issues. Close and reopen VS Code to see if IntelliSense starts working.
Update VS Code:
Ensure you’re using the latest version of VS Code. Updates often include bug fixes and improvements.
Select Correct Programming Language Mode:
Make sure you’re in the correct language mode. If you’re working with a specific language, select it from the bottom-right corner of the VS Code

**Reply by wintertims** [reply]
Thanks @monkeylander3000 . The "Make sure you’re in the correct language mode. If you’re working with a specific language, select it from the bottom-right corner of the VS Code." was the issue. Cheers mate

**Reply by monkeylander3000** [reply]
No problem! If you encounter any more issues, feel free to ask. Cheers! 😊

**Reply by Aspecky** [reply]
I am still running into this issue, syntax highlighting works and so does the debugger, but IntelliSense and formatter dont

**Reply by mark-wiemer** [reply]
Hey @Aspecky, can you provide details and screenshots? What solutions have you tried?
